docs/authentication.md

```diff
@@ -0,0 +1,134 @@
+            * Used for registering a certificate for Ledger and Auditor to verify a client-generated signature.
+        * `scalar.dl.client.entity.identity.digital_signature.cert_version` (or `scalar.dl.client.cert_version`, which is deprecated.)
+        * `scalar.dl.client.entity.identity.digital_signature.private_key_pem` or `scalar.dl.client.entity.identity.digital_signature.private_key_path` (or `scalar.dl.client.private_key_pem` or `scalar.dl.client.private_key_path`, which are deprecated.)
+            * Used for signing a request.
+    * Ledger-side properties
+      * `scalar.dl.ledger.authentication.method` (set "digital-signature")
+    * Auditor-side properties
+      * `scalar.dl.auditor.authentication.method` (set "digital-signature")
+* Ledger authentication 
+    * Ledger-side properties
+        * `scalar.dl.ledger.proof.enabled` (set true)
+            * Required because Ledger authentication uses the signatures of asset proofs.
+        * `scalar.dl.ledger.proof.private_key_pem` or `scalar.dl.ledger.proof.private_key_path`  
+            * Used for signing asset proofs.
+    * Auditor-side properties
+        * `scalar.dl.auditor.ledger.cert_holder_id`
+        * `scalar.dl.auditor.ledger.cert_version`
+            * Used for verifying the signatures of asset proofs.
+* Auditor authentication
+    * Ledger-side properties
+        * `scalar.dl.ledger.auditor.cert_holder_id`
+          * Used for verifying an Auditor-generated signature attached to a client request.
+        * `scalar.dl.ledger.auditor.cert_version`
+          * Used for verifying an Auditor-generated signature attached to a client request.
+    * Auditor-side properties
+        * `scalar.dl.auditor.cert_holder_id`
+          * Used for calling Ledger services directly by using the client library.
+        * `scalar.dl.auditor.cert_version`
+          * Used for calling Ledger services directly by using the client library.
+        * `scalar.dl.auditor.private_key_pem` or `scalar.dl.auditor.private_key_path`
+          * Used for signing a client request and a request from Auditor to Ledger.
+
+### HMAC
+
+* Client authentication
+    * Client-side properties
+        * `scalar.dl.client.authentication_method` (set "hmac")
+        * `scalar.dl.client.entity.id`
+            * Used for identifying a client.
+        * `scalar.dl.client.entity.identity.hmac.secret_key`
+            * Used for signing a request.
+        * `scalar.dl.client.entity.identity.hmac.key_version`
+    * Ledger-side properties
+      * `scalar.dl.ledger.authentication.method` (set "hmac")
+    * Auditor-side properties
+      * `scalar.dl.auditor.authentication.method` (set "hmac")
+
+* Ledger and Auditor authentication
+    * Ledger-side properties
+        * `scalar.dl.ledger.proof.enabled` (set true)
+            * Required because Ledger authentication uses the signatures of asset proofs.
+        * `scalar.dl.ledger.authentication.hmac.cipher_key`
```

Comment:
When I set only `scalar.dl.ledger.authentication.hmac.cipher_key` but `scalar.dl.ledger.servers.authentication.hmac.secret_key` didn't, I got a following exception:
```java
java.lang.IllegalArgumentException: Authentication between Ledger and Auditor is not correctly configured.Set scalar.dl.ledger.servers.authentication.hmac.secret_key or set a private key with scalar.dl.ledger.proof.private_key_path or scalar.dl.ledger.proof.private_key_pem.
        at com.scalar.dl.ledger.config.LedgerConfig.load(LedgerConfig.java:409)
...
```
Then, I set the `scalar.dl.ledger.servers.authentication.hmac.secret_key` to the ledger (and Auditor) configuration, then I can execute a contract successfully.

So I think `scalar.dl.ledger.servers.authentication.hmac.secret_key`(and `scalar.dl.auditor.servers.authentication.hmac.secret_key`) is also required if we use HMAC authenticaion. However there is no indication in this document.

Then I suggest adding the information that `scalar.dl.ledger.servers.authentication.hmac.secret_key` is needed to this document.